filebeat/harvester/source/source.go

```diff
@@ -15,3 +17,12 @@ type FileSource interface {
 	Stat() (os.FileInfo, error)
 	Continuable() bool // can we continue processing after EOF?
 }
+
+func NewFile(file *os.File, compression string) (FileSource, error) {
+
+	if config.GZipCompression == compression {
```

Comment:
We should definitively extract it and potentially abstract it if we introduce further compressions. But for now I think it is ok as long as we only have 1 special case. My usual approach is to start abstracting as soon as there are > 2 specific implementation.